### DISCUSSION TITLE
Do we need to React.lazy import screen components?

### DISCUSSION THREAD

**Discussion by anhnch** [react-navigation/react-navigation#11057] [Q&A] [answered]
Do we need to use React.lazy to dynamically import the screen components? or React Navigation does it under the hood?

Is this necessary?
```typescript
const ProfileScreen = React.lazy(() => import('./ProfileScreen'));
const SettingsScreen = React.lazy(() => import('./SettingsScreen'));
...
<Stack.Screen name="Profile" component={ProfileScreen} />
<Stack.Screen name="Settings" component={SettingsScreen} />
```

And in this scenario. The tabs are lazily loaded by React Navigation, but without React.lazy it has to import all stacks, am I right?
```typescript
import DefaultStack from './DefaultStack';
const ProjectStack = React.lazy(() => import('./ProjectStack'));

<Tabs.Navigator initialRouteName="DefaultStack" lazy={true}>
    <Tabs.Screen name='DefaultStack' component={DefaultStack} />
    <Tabs.Screen name='ProjectStack' component={ProjectStack} />
    ...
</Tabs.Navigator>
```

**Comment by satya164** [comment, marked as answer]
If you're using React Navigation on Web and you want to lazyload those screens, then yes, you should use `React.lazy` to import them.

It's not necessary for Android & iOS, but you can use `getComponent` to lazily evaluate the module: https://reactnavigation.org/docs/screen/#getcomponent

**Reply by anhnch** [reply]
Thank you for the information. It answers all of my questions

**Comment by jayshah123** [comment]
@satya164 `getComponent` supports synchronous require, will it be possible to use async dynamic import?
Starting with RN 73, metro supports lazy bundling with dynamic imports - https://metrobundler.dev/docs/module-api/#import-dynamic-import

**Reply by satya164** [reply]
@jayshah123 `getComponent` is just another option. You can always use `React.lazy` instead if you want asynchronous loading. Though you may want to specify a suspense boundary to handle loading placeholder in such cases https://reactnavigation.org/docs/7.x/navigator#screen-layout
